### PR TITLE
Tharsheblows error duplicate value update

### PR DESCRIPTION
**Pull request by rachelbaker**
Replaces #2800 with corrected code formatting.

### CODE REVIEW

**Comment by rachelbaker** [comment]
@WP-API/amigos #reviewmerge please

File: lib/fields/class-wp-rest-meta-fields.php [hunk 1]
```diff
@@ -222,7 +222,19 @@ protected function update_meta_value( $object, $name, $value ) {
...
 		}
 
-		if ( ! update_metadata( $this->get_meta_type(), $object, wp_slash( $name ), wp_slash( $value ) ) ) {
+		$meta_type  = $this->get_meta_type();
+		$meta_key   = wp_slash( $name );
+		$meta_value = wp_slash( $value );
+
+		// Do the exact same check for a duplicate value as in update_metadata() to avoid update_metadata() returning false.
+		$old_value = get_metadata( $meta_type, $object, $meta_key );
+		if ( 1 === count( $old_value ) ) {
```
Comment (joehoyle) [inline lib/fields/class-wp-rest-meta-fields.php:231]: Does this need to be nested? Seems like it might be simpler as:

``` php
if ( 1 === count( $old_value ) && $old_value[0] === $meta_value ) ...
```

Comment (joehoyle) [reply lib/fields/class-wp-rest-meta-fields.php:231]: What if the meta value has more than a single item? Does `update_metadata` not return `false` ever in that case for duplicate entries?

File: tests/test-rest-post-meta-fields.php [hunk 2]
```diff
@@ -230,6 +230,37 @@ public function test_set_value() {
 	}
 
 	/**
+	 * @depends test_get_value
+	 */
+	public function test_set_duplicate_single_value() {
```
Comment (joehoyle) [inline tests/test-rest-post-meta-fields.php:235]: Do we need to test duplicate with multiple, or is that covered in another test / PR?

Comment (rachelbaker) [reply tests/test-rest-post-meta-fields.php:235]: @joehoyle `update_metadata()` only does the duplicate check for single meta keys, for multiples it just goes ahead and performs the update.

File: lib/fields/class-wp-rest-meta-fields.php (hunk 1 above)
Comment (rachelbaker) [reply lib/fields/class-wp-rest-meta-fields.php:231]: @joehoyle No, `update_metadata()` only returns `false` if you are updating a single meta value and the previous and new values are the same.

Comment (rachelbaker) [reply lib/fields/class-wp-rest-meta-fields.php:231]: @joehoyle RE: the simpler conditional... technically yes (although a micro-optimization at best) - but the structure used by @tharsheblows here DOES indeed match the conditional structure already in `update_metadata()`

**Comment by joehoyle** [comment]
Sounds good!

**Review by joehoyle — APPROVED** [review]
